src/gui/virtualstudio.cpp

```diff
@@ -643,56 +635,27 @@ void VirtualStudio::completeConnection()
     emit connectionStateChanged();
     VsServerInfo* studioInfo = static_cast<VsServerInfo*>(m_servers.at(m_currentStudio));
     try {
-        m_jackTrip.reset(new JackTrip(JackTrip::CLIENTTOPINGSERVER, JackTrip::UDP, 2, 2,
-#ifdef WAIR  // wair
-                                      0,
-#endif  // endwhere
-                                      4, 1));
-        m_jackTrip->setConnectDefaultAudioPorts(true);
-#ifdef RT_AUDIO
-        if (m_useRtAudio) {
-            m_jackTrip->setAudiointerfaceMode(JackTrip::RTAUDIO);
-            m_jackTrip->setSampleRate(studioInfo->sampleRate());
-            m_jackTrip->setAudioBufferSizeInSamples(m_bufferSize);
-            if (m_inputDevice == QLatin1String("(default)")) {
-                m_jackTrip->setInputDevice("");
-            } else {
-                m_jackTrip->setInputDevice(m_inputDevice.toStdString());
-            }
-            if (m_outputDevice == QLatin1String("(default)")) {
-                m_jackTrip->setOutputDevice("");
-            } else {
-                m_jackTrip->setOutputDevice(m_outputDevice.toStdString());
-            }
+        std::string input = m_inputDevice.toStdString();
+        if (m_inputDevice == QLatin1String("(default)")) {
+            input = "";
         }
-#endif
-        m_jackTrip->setBufferStrategy(1);
-        m_jackTrip->setBufferQueueLength(-500);
-        m_jackTrip->setPeerAddress(studioInfo->host());
-        m_jackTrip->setPeerPorts(studioInfo->port());
-        m_jackTrip->setPeerHandshakePort(studioInfo->port());
+        std::string output = m_outputDevice.toStdString();
```

Comment:
You need to wrap any reference to `m_outputDevice` and `m_inputDevice` in an `#ifdef RT_AUDIO`

Reply:
basically all of these from `virtualstudio.h`

`QStringList m_inputDeviceList;
    QStringList m_outputDeviceList;
    QString m_inputDevice;
    QString m_outputDevice;
    quint16 m_bufferSize;
    QString m_previousInput;
    QString m_previousOutput;
    quint16 m_previousBuffer;
    bool m_previousUseRtAudio = false;`